auto_submit/lib/service/validation_service.dart

```diff
@@ -288,4 +282,14 @@ extension PullRequestExtension on github.PullRequest {
     final slug = base!.repo!.slug();
     return mqEnabledRepos.contains(slug.fullName);
   }
+
+  /// Extracts label names from the `IssueLabel` list [labels].
+  List<String> get labelNames {
+    final labels = this.labels;
+    if (labels == null) {
+      return const <String>[];
+    }
```

Comment:
nit:

```dart
if (labels == null) return const <String>[];
return labels.map<String>((github.IssueLabel labelMap) => labelMap.name).toList();
```

or 

```dart
return labels?.map<String>((github.IssueLabel labelMap) => labelMap.name).toList() ?? <String> [];
```

Reply:
The compiler cannot promote `this.labels` to non-nullable type without a local variable. The `foo?...... ?? defaultValue` trick works and is clever, but I'm not sure if it's readable 🤔

Reply:
sgtm